src/otx/algo/visual_prompting/openvino_models.py

```diff
@@ -0,0 +1,151 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""Openvino Model Wrappers for the OTX visual prompting."""
+
+from __future__ import annotations
+
+from copy import deepcopy
+from typing import TYPE_CHECKING, Any
+
+import numpy as np
+from openvino.model_api.models import ImageModel, SegmentationModel
+from openvino.model_api.models.types import BooleanValue, NumericalValue, StringValue
+
+if TYPE_CHECKING:
+    from openvino.model_api.adapters.inference_adapter import InferenceAdapter
+
+
+class ImageEncoder(ImageModel):
+    """Image Encoder class of OTX Visual Prompting model for openvino task."""
+
+    __model__ = "image_encoder"
+
+    def __init__(
+        self,
+        inference_adapter: InferenceAdapter,
+        configuration: dict[str, Any] | None = None,
+        preload: bool = False,
+    ):
+        super().__init__(inference_adapter, configuration, preload)
+
+    @classmethod
+    def parameters(cls) -> dict[str, Any]:  # noqa: D102
+        parameters = super().parameters()
+        parameters.update(
+            {
+                "resize_type": StringValue(default_value="fit_to_window"),
+                "image_size": NumericalValue(value_type=int, default_value=1024, min=0, max=2048),
+            },
+        )
+        return parameters
+
+    def preprocess(self, inputs: np.ndarray) -> tuple[dict[str, np.ndarray], dict[str, Any]]:
+        """Update meta for image encoder."""
+        dict_inputs, meta = super().preprocess(inputs)
+        meta["resize_type"] = self.resize_type
+        return dict_inputs, meta
+
+
+class Decoder(SegmentationModel):
```

Comment:
```suggestion
class VisualPromptDecoder(SegmentationModel):
```

Reply:
I updated ov models' names as you recommended, `Decoder` -> `VisualPromptingDecoder` and `ImageEncoder` -> `VisualPromptingImageEncoder` at https://github.com/openvinotoolkit/training_extensions/pull/2955/commits/c7f5f1e611d4503b368ab506c9dd0ea86ffe4518.